torch/_subclasses/functional_tensor.py

```diff
@@ -428,10 +433,17 @@ def unwrap(x):
             assert not torch._C._dispatch_has_kernel_for_dispatch_key(
                 func.name(), torch._C.DispatchKey.Functionalize
             )
-            return handle_effects(
-                self._allow_token_discovery, self._tokens, func, args, kwargs
+            from torch._functorch._aot_autograd.traced_function_transforms import (
+                set_partitioner_tag,
```

Comment:
Right now you are directly using this context manager here in the `FunctionalTensor.__torch_dispatch__` code, because you are trying to carefully set the meta field for **only** any `with_effects()` ops that are called in the backward.

It feels bit bad that `FunctionalTensor` has to know about this code. Instead, what if we:

(1) agree that **all** nodes that are invoked as part of the `autograd.grad()` part of the joint function will get this metadata (basically just a bool that specifies "is_backward")

(2) the partitioner will just have access to this metadata for all backward nodes (even if it only really needs it for mutable ops / effectful ops)